### ISSUE TITLE
Assignability between number and enum type

### ISSUE THREAD

**Issue by rkirov** [Microsoft/TypeScript#10738]
**TypeScript Version:**  nightly (2.0.0-dev.201xxxxx)

**Code**

```ts
enum E {
  A,
  B
}

let n: number = 0;

let y: E = n;
let z: E|boolean = n;  // ERROR(9,5): : Type 'number' is not assignable to type 'boolean | E'.

```

**Expected behavior:**

Either disallow assigning `number` to `E`, reasonable if one considers `E` equivalent to `0 | 1`. Or allow the assignment of `number` to `E | boolean`.

**Actual behavior:** 

let z: E|boolean = n;  // ERROR(9,5): : Type 'number' is not assignable to type 'boolean | E'.

**Comment by RyanCavanaugh** [comment]
Regression from 1.8 behavior

**Comment by rkirov** [comment]
Also for added mystery, changing the enum to single element `enum E {A}` removes the error. Admittedly, single member enums are not very common, but it might help you guys when debugging.

**Comment by sandersn** [comment]
I think this is caused by the switch to enums-as-unions, which would explain why the single-member enum doesn't have the bug.

**Comment by sandersn** [comment]
Also `let z: E | boolean = 0` is fine too, so I think the `0 | 1` elements of the enum are incorrectly not assignable to `number`.

**Comment by sandersn** [comment]
`isSimpleTypeRelatedTo` has a special case that lets numbers be assignable to enums. Looks like it (or related code) should also have a special case for union types that contain enums.

**Comment by sandersn** [comment]
Fix is up at #10784
